### PR TITLE
openstack_db_configuration_v1

### PR DESCRIPTION
**Pull request by eugenetaranov**
This PR adds openstack_db_configuration_v1 resource and addresses https://github.com/terraform-providers/terraform-provider-openstack/issues/141

This is a contribution from [Ribose Inc.](https://www.ribose.com)

### CODE REVIEW

**Comment by jtopjian** [comment]
@eugenetaranov Can you have two separate PRs for this? One for the resource and one for the addition to the `openstack_db_instance_v1` resource?

**Comment by eugenetaranov** [comment]
@jtopjian removed all references to openstack_db_instance_v1 resource, please review

**Comment by ronaldtse** [comment]
Thank you @eugenetaranov 👍

**Comment by ronaldtse** [comment]
Thank you @eugenetaranov 👍

File: openstack/resource_openstack_db_configuration_v1.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceDatabaseConfigurationV1() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDatabaseConfigurationV1Create,
+		Read:   resourceDatabaseConfigurationV1Read,
+		Delete: resourceDatabaseConfigurationV1Delete,
+		Importer: &schema.ResourceImporter{
```
[the remark this answers is not in the document]
Comment (eugenetaranov) [reply openstack/resource_openstack_db_configuration_v1.go]: Debugged import feature, looks like it partially works, i.e. imports name, description and region but the rest of the attributes is lost. So removed import for now.

File: openstack/resource_openstack_db_configuration_v1.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"datastore": &schema.Schema{
```
[the remark this answers is not in the document]
Comment (eugenetaranov) [reply openstack/resource_openstack_db_configuration_v1.go]: thanks, fixed

File: openstack/resource_openstack_db_configuration_v1_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+	"github.com/gophercloud/gophercloud/openstack/db/v1/configurations"
+)
+
+func TestAccDatabaseV1Configuration_basic(t *testing.T) {
+	var configuration configurations.Config
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheckDatabase(t) },
+		Providers: testAccProviders,
```
[the remark this answers is not in the document]
Comment (eugenetaranov) [reply openstack/resource_openstack_db_configuration_v1_test.go]: fixed

**Comment by eugenetaranov** [comment]
@jtopjian please review

**Comment by jtopjian** [comment]
LGTM! Thank you for your work on this :smile:

**Pull request merged**